### PR TITLE
cache metadata in redis

### PR DESCRIPTION
**Pull request by supernothing**

### CODE REVIEW

File: src/polyswarmd/__main__.py [hunk 1]
```diff
@@ -15,7 +16,8 @@
 @click.option('--log-level', envvar='LOG_LEVEL', default='WARNING', help='Logging level')
 @click.option('--host', default='', help='Host to listen on')
 @click.option('--port', default=31337, help='Port to listen on')
-def main(log_format, log_level, host, port):
+@click.option('--redis-uri', default=None, envvar='REDIS', help='Redis server to use for caching')
+def main(log_format, log_level, host, port, redis_uri):
```
[the remark this answers is not in the document]
Comment (supernothing) [reply src/polyswarmd/__main__.py]: No.

File: src/polyswarmd/artifacts.py [hunk 2]
```diff
@@ -79,16 +79,34 @@ def post_to_ipfs(files, wrap_dir=False):
...
+    h = json.loads(r.text.splitlines()[-1])['Hash']
 
+    if cache and config.redis:
+        config.redis.set(f'polyswarmd:{h}', files[0][1], ex=300)
 
-def get_from_ipfs(ipfs_uri, ipfs_root=None, session=None):
+    return 201, h
+
+
+def get_from_ipfs(ipfs_uri, ipfs_root=None, session=None, redis=None):
```
[the remark this answers is not in the document]
Comment (supernothing) [reply src/polyswarmd/artifacts.py]: shut

**Comment by supernothing** [comment]
Currently deployed on stage, seems to have solved our IPFS issues.

**Comment by Rizato** [comment]
I wish this was done in my bid per artifact branch. There are a lot of changes this will have to be translated to.

File: src/polyswarmd/config.py [hunk 3]
```diff
@@ -310,8 +312,10 @@ def from_config_file(cls, filename):
         require_api_key = auth_uri is not None
         trace_transactions = config.get('trace_transactions', True)
         profiler_enabled = config.get('profiler_enabled', False)
+        redis_uri = config.get('redis_uri', os.environ.get('REDIS_URI', None))
```
Comment (Rizato) [inline src/polyswarmd/config.py]: I'm not a fan of doing it via environment fallback. 
I think we should remain consistent and just do it via file/consul.

Comment (Rizato) [reply src/polyswarmd/config.py]: This is consistent with AUTH_URI

Comment (supernothing) [reply src/polyswarmd/config.py]: Yeah, I was copying from `AUTH_URI` there. And I agree, it has caught both @aarongooch and I before thinking setting an env var would make polyswarmd do something, but I think we should do a PR later that removes all these at once.

**Pull request merged**